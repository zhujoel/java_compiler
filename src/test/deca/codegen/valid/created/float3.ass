; Test pour savoir si la pile est pleine
	TSTO #6
	BOV pile_pleine
	ADDSP #4
; start main program
; Class Declaration
; Code de la table des méthodes de Object
	LOAD #null, R0
	STORE R0, 4(GB)
	LOAD code.Object.equals, R0
	STORE R0, 5(GB)
; Main Function
; Declaration des variables
	LOAD #0x1.0p0, R2
	STORE R2, 9(GB)
; Beginning of main instructions:
; *
	LOAD 9(GB), R2
	LOAD #0x1.b33334p1, R3
	MUL R2, R3
	BOV debordement
	STORE R3, 9(GB)
	LOAD 9(GB), R1
	WFLOAT
	HALT
<<<<<<< HEAD
=======
code.Object.equals:
>>>>>>> c423b6d2167ba364dc50207abb16cf5dd4ebba6e
pile_pleine:
	WSTR "Erreur : Pile pleine."
	WNL
	ERROR
dereferencement_null:
	WSTR "Erreur : dereferencement null."
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : Tas plein."
	WNL
	ERROR
debordement:
	WSTR "Erreur : debordement durant une operation arithmétique"
	WNL
	ERROR
io_error:
	WSTR "Erreur : input ou output"
	WNL
	ERROR
; end main program
