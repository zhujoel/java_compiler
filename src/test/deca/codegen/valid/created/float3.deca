{
    float a = 1.0;
    a = a * 3.4;
    print(a);
}
