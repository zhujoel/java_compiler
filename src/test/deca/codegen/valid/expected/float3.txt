3.40000e+00
